packages/hoppscotch-common/src/helpers/backend/helpers.ts

```diff
@@ -272,3 +274,27 @@ export const getTeamCollectionJSON = async (teamID: string) => {
   const hoppCollections = collections.map(teamCollectionJSONToHoppRESTColl)
   return E.right(JSON.stringify(hoppCollections, null, 2))
 }
+
+/**
+ * Verifies if the current user's authentication tokens are valid
+ * @returns Promise<boolean> - true if tokens are valid, false otherwise
+ */
+export const verifyAuthTokens = async (): Promise<boolean> => {
+  try {
+    const BACKEND_API_URL = import.meta.env.VITE_BACKEND_API_URL
+
+    const response = await axios.get(`${BACKEND_API_URL}/auth/verify-token`, {
+      headers: {
+        "Content-Type": "application/json",
+        ...platform.auth.getBackendHeaders(),
+      },
+      withCredentials: true,
+    })
+
+    // axios automatically throws on error status codes, so if we reach here, it was successful
+    return !!response.data.isValid
+  } catch (error) {
+    console.debug("Token verification failed:", error)
+    return false
+  }
+}
```

Comment:
Given this is exclusive to the SH offering, we'll need to abstract this behind a platform definition to ensure the cloud offering (which uses Firebase's automatic token refresh) remains unaffected.

For instance, `platform.auth.validateUserSession()` defaults to `true` in platforms where it isn't defined.